cli/src/main/java/com/devonfw/tools/ide/context/GitUrl.java

```diff
@@ -11,6 +13,76 @@
  */
 public record GitUrl(String url, String branch) {
 
+  // Regex pattern to match SSH URLs
+  private static final Pattern SSH_URL_PATTERN = Pattern.compile("git@([a-zA-Z0-9._-]+):([a-zA-Z0-9._/-]+)\\.git", Pattern.CASE_INSENSITIVE);
```

Comment:
It is fragile to make assumptions about these segments for host, etc.
At least for the first segment/group there is a simple solution:
```suggestion
  private static final Pattern SSH_URL_PATTERN = Pattern.compile("git@([^:]+):([a-zA-Z0-9._/-]+)\\.git", Pattern.CASE_INSENSITIVE);
```